plugins/inputs/tencentcloudcm/products.go

```diff
@@ -0,0 +1,387 @@
+package tencentcloudcm
+
+import (
+	"encoding/json"
+	"fmt"
+
+	cdb "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/cdb/v20170320"
+	clb "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/clb/v20180317"
+	"github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common"
+	"github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common/profile"
+	cvm "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/cvm/v20170312"
+	dc "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/dc/v20180410"
+	es "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/es/v20180416"
+	redis "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/redis/v20180412"
+)
+
+// Product defines cloud product
+type Product interface {
+	Namespace() string // Tencent Cloud CM Product Namespace
+	Metrics() []string // Supported metrics
+	Keys() []string    // Product Dimension Key fields
+	Discover(crs *common.Credential, region, endpoint string, offset, limit int64) (uint64, []map[string]interface{}, error)
+}
+
+// DC defines Direct Connect, see: https://intl.cloud.tencent.com/document/product/216
+type DC struct{}
+
+// Namespace implements Product interface
+func (d DC) Namespace() string {
+	return "QCE/DC"
+}
+
+// Metrics implements Product interface
+func (d DC) Metrics() []string {
+	// see: https://intl.cloud.tencent.com/document/product/248/10994
+	return []string{"OutBandwidth", "InBandwidth"}
+}
+
+// Keys implements Product interface
+func (d DC) Keys() []string {
+	return []string{"directConnectId"}
+}
+
+// Discover implements Product interface
+func (d DC) Discover(crs *common.Credential, region, endpoint string, offset, limit int64) (uint64, []map[string]interface{}, error) {
+	cpf := profile.NewClientProfile()
+	cpf.HttpProfile.Endpoint = fmt.Sprintf("%s.%s", "dc", endpoint)
+	client, err := dc.NewClient(crs, region, cpf)
+	if err != nil {
+		return 0, nil, fmt.Errorf("dc.NewClient failed, error: %s", err)
+	}
+
+	request := dc.NewDescribeDirectConnectTunnelsRequest()
+	request.Offset = common.Int64Ptr(offset)
+	request.Limit = common.Int64Ptr(limit)
+
+	response, err := client.DescribeDirectConnectTunnels(request)
+	if err != nil {
+		return 0, nil, fmt.Errorf("discover instances for namespace %s failed: %s", d.Namespace(), err)
+	}
+
+	instances := []map[string]interface{}{}
```

Comment:
Usually you see
```suggestion
	var instances []map[string]interface{}
```
Same for all following `Discover()` instances.

Reply:
fixed in commit 880ecd1